src/TopPage/FluentExtension.php

```diff
@@ -39,10 +39,36 @@ protected function assignTopPage(Page $page): void
         $this->owner->TopPageLocale = FluentState::singleton()->getLocale();
     }
 
+    /*
+     * @inheritdoc
+     */
     protected function clearTopPage(): void
     {
         parent::clearTopPage();
 
         $this->owner->TopPageLocale = null;
     }
+
+    /*
+     * @inheritdoc
+     */
+    protected function assignFixedTopPage(): void
+    {
+        parent::assignFixedTopPage();
+
+        $this->owner->TopPageLocale = FluentState::singleton()->getLocale();
+    }
+
+    /*
+     * @inheritdoc
+     */
+    protected function saveChanges(array $extraData = []): void
+    {
+        /** @var DataObject|FluentExtension $owner */
+        $owner = $this->owner;
+
+        parent::saveChanges([
+            '"TopPageLocale"' => $owner->TopPageLocale,
```

Comment:
You need to get the `$extraData` param here passed to DataExtension::saveChanges()